### PR TITLE
Operator config controller should utilize POD_NAME env var for Deployment lookup

### PR DESCRIPTION
**Pull request by dmvolod**

### CODE REVIEW

**Review by ryanemerson — APPROVED** [review]

File: pkg/kubernetes/pods.go [hunk 1]
```diff
@@ -53,3 +62,45 @@ func GetEnvVarIndex(envVarName string, env *[]corev1.EnvVar) int {
...
+func GetOperatorPodOwnerRef(ns string, client crclient.Client) (*metav1.OwnerReference, error) {
+	// Get current Pod the operator is running in
+	pod, err := k8sutil.GetPod(context.TODO(), client, ns)
+	if err != nil {
+		return nil, err
+	}
+	podOwnerRefs := metav1.NewControllerRef(pod, pod.GroupVersionKind())
+	// Get Owner that the Pod belongs to
+	ownerRef := metav1.GetControllerOf(pod)
+	finalOwnerRef, err := findFinalOwnerRef(ns, ownerRef, client)
```
Comment (rigazilla) [inline pkg/kubernetes/pods.go:75]: When installed via OLM the CSV owns the Deployment. Does this produce any drawback? At first sight seems not...

Comment (dmvolod) [reply pkg/kubernetes/pods.go:75]: This code ported from the Operator SDK which is using for Service owner reference. Not sure that's will not work for Grafana Dashboard

Comment (dmvolod) [reply pkg/kubernetes/pods.go:75]: And we don't another alternatives way if Deployment name has been changed without labels which will be supported at least in the OCP 4.10
https://github.com/operator-framework/operator-lifecycle-manager/pull/2054

Comment (rigazilla) [reply pkg/kubernetes/pods.go:75]: It seems to me that Csv and Deployment have the same lifespan, so I think it's almost the same to have one or the other as owner for the dashboard object.

**Review by rigazilla — APPROVED** [review]

**Comment by dmvolod** [comment]
I think we can merge this PR
Thanks @ryanemerson and @rigazilla for review

**Pull request merged**